src/tool_writeout_json.c

```diff
@@ -56,18 +56,22 @@ static const struct mapping mappings[]={
   {"http_code", JSON_LONG, CURLINFO_RESPONSE_CODE},
   {"response_code", JSON_LONG, CURLINFO_RESPONSE_CODE},
   {"http_connect", JSON_LONG, CURLINFO_HTTP_CONNECTCODE},
+#ifndef DEBUGBUILD
```

Comment:
Unconditional filtering based on this symbol is not desirable. When people (like me) build curl debug-enabled it should still work like normally by default. A debug build can however check for example an environment variable and do special magic based on that, which the release build shouldn't. Then the test case can set that env variable but when you run curl in a shell and still have full json functionality...

Reply:
Ok thanks for the "tutorial". Will rework this solution today or tomorrow depending if I find time.